File: library/src/main/java/com/github/daneko/android/iab/model/ActivityResults.java

package com.github.daneko.android.iab.model;

import android.content.Intent;

import fj.data.Option;

import lombok.Value;

/**
 * {@link android.app.Activity#onActivityResult(int, int, android.content.Intent)} のパラメータをまとめたもの
 */
@Value(staticConstructor = "of")
public class ActivityResults {
    int requestCode;
    int resultCode;
    Option<Intent> data;
}

File: library/src/main/kotlin/com/github/daneko/android/iab/model/ActivityResults.kt
package com.github.daneko.android.iab.model

import android.content.Intent
import fj.data.Option


/**
 * [android.app.Activity.onActivityResult] のパラメータをまとめたもの
 */
data class ActivityResults(
        val requestCode: Int,
        val resultCode: Int,
        val data: Option<Intent> = Option.none()) {

    companion object {
        @JvmStatic @JvmOverloads
        fun of(requestCode: Int, resultCode: Int, data: Option<Intent> = Option.none()):ActivityResults {
            return ActivityResults(requestCode, resultCode, data)
        }
    }
}
